barretenberg/cpp/src/barretenberg/common/std_array.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <array>
+
+/**
+ * @brief Concatenates multiple std::array objects into a single array.
+ *
+ * This function template takes a variadic number of std::array objects and concatenates
+ * their elements into a single std::array. The size of the resulting array is the sum of the sizes
+ * of the input arrays.
+ *
+ * @tparam T The type of elements stored in the arrays.
+ * @tparam Ns The sizes of the input arrays. This is a variadic template parameter pack representing
+ *            the sizes of each input array.
+ * @param arrays Variadic number of std::array objects to concatenate. Each array can have a
+ *               different size but must contain the same type of elements.
+ * @return std::array<T, (Ns + ...)> A new std::array containing all elements from the input arrays
+ *         concatenated in the order they were passed.
+ *
+ * Example usage:
+ *   std::array<int, 2> a = {1, 2};
+ *   std::array<int, 3> b = {3, 4, 5};
+ *   auto result = concatenate(a, b);  // result is std::array<int, 5>{1, 2, 3, 4, 5}
+ */
+template <typename T, std::size_t... Ns> std::array<T, (Ns + ...)> concatenate(const std::array<T, Ns>&... arrays)
+{
+    std::array<T, (Ns + ...)> result;
+
```

Comment:
Concatenation helper for std::array. I decided to call all of these 'concatenate'. I removed a redundant helper. The other helper was more dense, but I found this one easier to follow

Reply:
This should have a namespace to not pollute users of bb as libraries (which may be other components we right, good to be clean). I have added https://github.com/AztecProtocol/barretenberg/issues/794. My preference is to hold off on doing this for when all the flavor classes are also in `bb`

Reply:
It could be. If it's smart, it'll realize this is an opportunity for memcpy. But so should the loop compilation.

I'll refactor for cleanliness though